ansi/sixel/encoder.go

```diff
@@ -0,0 +1,270 @@
+package sixel
+
+import (
+	"fmt"
+	"image"
+	"image/color"
+	"io"
+	"strconv"
+	"strings"
+
+	"github.com/bits-and-blooms/bitset"
+)
+
+// Sixels are a protocol for writing images to the terminal by writing a large blob of ANSI-escaped data.
+// They function by encoding columns of 6 pixels into a single character (in much the same way base64
+// encodes data 6 bits at a time). Sixel images are paletted, with a palette established at the beginning
+// of the image blob and pixels identifying palette entires by index while writing the pixel data.
+//
+// Sixels are written one 6-pixel-tall band at a time, one color at a time. For each band, a single
+// color's pixels are written, then a carriage return is written to bring the "cursor" back to the
+// beginning of a band where a new color is selected and pixels written. This continues until the entire
+// band has been drawn, at which time a line break is written to begin the next band.
+
+const (
+	LineBreak        byte = '-'
+	CarriageReturn   byte = '$'
+	RepeatIntroducer byte = '!'
+	ColorIntroducer  byte = '#'
+	RasterAttribute  byte = '"'
+)
+
+type Options struct {
+}
+
+type Encoder struct {
+}
+
+func Raster(pan, pad, ph, pv int) string {
+	return fmt.Sprintf("%s%d;%d;%d;%d", string(RasterAttribute), pan, pad, ph, pv)
+}

```

Comment:
Perhaps we can have a `type Raster struct { Pan, Pad, Ph, Pv int}` with `func (Raster) Encode(io.Writer) error`, `func (Raster) String() string` that uses `r.Encode`, and `func ParseRaster([]byte) (Raster, error)`